wasm/ADDRESS.wast

```diff
@@ -0,0 +1,23 @@
+(func $ADDRESS
+  (param $sp i32)
+  (local $scratch i32)
+  (result i32)
+
+  ;; there's no input item for us to overwrite
+  (set_local $sp (i32.add (get_local $sp) (i32.const 32)))
+
+  (set_local $scratch (i32.const 32776))
+
+  ;; clear the 3rd 64bit chunk so we don't need to mask it later
+  (i64.store (i32.add (get_local $scratch) (i32.const 16)) (i64.const 0))
+
+  ;; loads the caller into memory
+  (call_import $address (get_local $scratch))
```

Comment:
I was going to keep it identical to CALLER, come up with a templateng system or something to replace all these state getters *and* after that do the optimisation step.